hr_timesheet_task_change_project/models/account_analytic_line.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = "account.analytic.line"
+
+    def _check_task_project(self):
+        # in the hr_timesheet module, this method is a constraint on task_id
+        # and project_id to ensure that the task to which the account analytic
+        # line is linked is linked to the same project. this override removes
+        # this constraint.
+        pass
+
+    def onchange_project_id(self):
+        # in the hr_timesheet module, this method sets task_id to False if the
+        # selected project is not the task's project. this override removes
+        # this behavior.
+        task_id = self.task_id
+        # the parent method returns a domain to limit the available tasks, so
+        # it needs to be returned.
+        result = super()
```

Comment:
this seems to work, but shouldn’t it be:
```suggestion
        result = super().onchange_project_id()
```

Reply:
En effet, j'ai essayé de comprendre pourquoi super() marchait tout seul, mais sans succès. Je dirai que ce serait une bonne idée d'utiliser `super().onchange_project_id()` car c'est ce qui est fait partout.